lib/auth/init.go

```diff
@@ -786,14 +787,15 @@ func initializeAuthPreference(ctx context.Context, asrv *Server, newAuthPref typ
 		}
 
 		if !shouldReplace {
-			if err := modules.ValidateResource(storedAuthPref); err != nil {
+			if os.Getenv(teleport.EnvVarAllowNoSecondFactor) != "true" {
+				err := modules.ValidateResource(storedAuthPref)
```

Comment:
I understand this is a revert commit but it hurts me to see us compare with "true" here and compare again with "yes" inside modules.ValidateResource

Can we remove the condition here and only check the env var inside modules.ValidateResource and use strconv.ParseBool?

But i guess in effect we are accepting either "true" or "yes" here so I'll approve in case you just want to keep the clean revert